Python/tests/datatypes/test_timex_relative_convert.py

```diff
@@ -121,64 +121,53 @@ def test_datatypes_relativeconvert_datetime_yesterday():
 
 
 def test_datatypes_relativeconvert_daterange_this_week():
-    timex = Timex(timex='2017-W40')
-    today = datetime(2017, 9, 25)
+    today = datetime.now()
+    year, week_of_year = today.isocalendar()[0:2]
+    timex = Timex(timex="{}-W{}".format(year, week_of_year))
 
     assert TimexRelativeConvert.convert_timex_to_string_relative(timex, today) == 'this week'
 
 
 def test_datatypes_relativeconvert_daterange_next_week():
-    timex = Timex(timex='2017-W41')
-    today = datetime(2017, 9, 25)
+    today = datetime.now()
+    next_week_date = today + timedelta(days=7)
+    year, week_of_year = next_week_date.isocalendar()[0:2]
+    timex = Timex(timex="{}-W{}".format(year, week_of_year))
 
     assert TimexRelativeConvert.convert_timex_to_string_relative(timex, today) == 'next week'
 
 
 def test_datatypes_relativeconvert_daterange_last_week():
-    timex = Timex(timex='2017-W39')
-    today = datetime(2017, 9, 25)
-
-    assert TimexRelativeConvert.convert_timex_to_string_relative(timex, today) == 'last week'
-
-
-def test_datatypes_relativeconvert_daterange_this_week_2():
-    timex = Timex(timex='2017-W41')
-    today = datetime(2017, 10, 4)
-
-    assert TimexRelativeConvert.convert_timex_to_string_relative(timex, today) == 'this week'
-
-
-def test_datatypes_relativeconvert_daterange_next_week_2():
-    timex = Timex(timex='2017-W42')
-    today = datetime(2017, 10, 4)
-
-    assert TimexRelativeConvert.convert_timex_to_string_relative(timex, today) == 'next week'
-
-
-def test_datatypes_relativeconvert_daterange_last_week_2():
-    timex = Timex(timex='2017-W40')
-    today = datetime(2017, 10, 4)
+    today = datetime.now()
+    next_week_date = today - timedelta(days=7)
+    year, week_of_year = next_week_date.isocalendar()[0:2]
+    timex = Timex(timex="{}-W{}".format(year, week_of_year))

```

Comment:
As unit tests, it would be better to have the timex to be compared against be hardcoded as original. It eases comparison between platforms and turning them into common resources later.
Also, no need to always use `now`. We want to make sure that past dates also work and we should have tests specific for them.